docs/faq.rst

```diff
@@ -0,0 +1,54 @@
+
+The same applies to naming a passkey after registration.
+
+.. _yubikey-registration:
+
+Why can't I register my YubiKey or security key when passwordless login is turned on?
+-------------------------------------------------------------------------------------
+
+If you’re trying to use a ``YubiKey`` or a similar security key with passwordless login enabled, you may run into registration issues. This is because when passwordless login is enabled, only certain types of devices can be registered. This helps prevent confusion when a device doesn’t work during authentication.
+
+In this case, the credential created isn’t technically a passkey. It's what the `Web Authentication specification <https://www.w3.org/TR/webauthn-3/>`_ calls a `server-side credential <https://www.w3.org/TR/webauthn-3/#server-side-credential>`_. Unlike a true passkey, where the :term:`private key` is stored on the device, the private key here is encrypted and stored in the :term:`credential ID` field on the server. This approach is necessary because some security keys have limited memory and can only store a small number of credentials.
+
+So to use a security key for passwordless login, the user must provide at least a username. This allows the system to look up the credentials associated with that account. However, this process requires returning all credential IDs to an unauthenticated and untrusted site visitor. This complicates the authentication process and risks leaking information about the user and their registered passkeys.
+
+Due to these complexities, we decided not to support this functionality.
+
+However, it’s possible to register and use these security keys when passwordless login functionality is disabled. When used as a :term:`second-factor authentication`, the user is already partially authenticated so it’s much more straightforward to return credential IDs.
+
+If you are curious, the option used to restrict registration is done by setting the ``residentKey`` :term:`authenticator` selection criteria to required. See the specification for details: `authenticatorSelection.residentKey in the Web Authentication specification <https://www.w3.org/TR/webauthn-2/#dom-authenticatorselectioncriteria-residentkey>`_.
+
+You can still register and use these security keys when passwordless login is disabled. When used as a second-factor authentication method, the user is already partially authenticated, making it easier to return credential IDs securely.
+
+If you want more details, this restriction is done by setting the residentKey authenticator selection criteria to required. For more information, see `authenticatorSelection.residentKey in the Web Authentication specification <https://www.w3.org/TR/webauthn-2/#dom-authenticatorselectioncriteria-residentkey>`_.
```

Comment:
@activus-d this appears to be the same as above but in different words. Which one to keep?